### PR TITLE
Fix having empty classes in the dynamic sheet

### PR DESCRIPTION
**Pull request by HenriBeck**
As discussed in #267 
This should remove empty classes and remove depending on jss-compose and jss-nested.

### CODE REVIEW

File: src/createHoc.js [hunk 1]
```diff
@@ -146,9 +145,12 @@ export default (stylesOrCreator, InnerComponent, options = {}) => {
         })
       }
 
-      const sheet = dynamicSheet || staticSheet
       const defaultClasses = InnerComponent.defaultProps ? InnerComponent.defaultProps.classes : {}
-      const classes = {...defaultClasses, ...sheet.classes, ...userClasses}
+      const classes = {
+        ...defaultClasses,
+        ...compose(staticSheet.classes, dynamicSheet ? dynamicSheet.classes : {}),
```
Comment (kof) [inline src/createHoc.js]: if no dynamicSheet why calling compose at all?

Comment (HenriBeck) [reply src/createHoc.js]: Yeah, added that improvement locally, just forgot to push

File: src/compose.js [hunk 2]
```diff
@@ -21,32 +21,12 @@
...
-    // Add rules which are defined in `classes` but aren't in styles.
-    for (const name in classes) {
-      const className = styles[name]
-      if (!className) {
-        styles[name] = {composes: classes[name]}
-      }
-    }
+export default (staticClasses, dynamicClasses) => {
+  for (const name in dynamicClasses) {
+    staticClasses[name] = staticClasses[name]
```
Comment (kof) [inline src/compose.js]: you are mutating the classes object that you got from the static style sheet, we shouldn't mutate that

File: src/createHoc.js [hunk 3]
```diff
@@ -146,9 +145,15 @@ export default (stylesOrCreator, InnerComponent, options = {}) => {
         })
       }
 
-      const sheet = dynamicSheet || staticSheet
       const defaultClasses = InnerComponent.defaultProps ? InnerComponent.defaultProps.classes : {}
```
Comment (kof) [inline src/createHoc.js]: btw instead of empty object it can be undefined, spreading still works (learned it after I wrote it I suppose)

**Pull request merged**